backend/api/view/FriendsView.py

```diff
@@ -13,7 +13,13 @@ def valid_input(data):
     try:
         if (len(data["query"]) == 0 or
                 len(data["author"]["id"]) == 0 or
-                len(data["friend"]["id"]) == 0):
+                len(data["author"]["host"]) == 0 or
+                len(data["author"]["displayName"]) == 0 or
```

Comment:
I understand what's going on but I think it's better for cleaner code practices to instead of making really big logic statements, separate them into their own variables then compare.

For example, isValidAuthorInfo = ((len(data["author"]["id] == 0) or etc...)
Same idea for isValidFriendInfo

then do if(isValidAuthorInfo or isValidFriendInfo):